### PR TITLE
updated nvrtc

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes the broken CI where register usage exceeds the limit causing launch failure.
Few fixes:

1. fix option argument passed for CUDA driver api (include register cap which caused CI failure);
2. set correct `compile_to_sass` flag;
3. add verbose log when exporting ptx/cubin binaries;

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 1]
```diff
@@ -854,22 +879,35 @@ NvrtcFunction nvrtcCompile(
           options.size(),
           options.data(),
           option_vals.data()));
+
+      if (isDebugDumpEnabled(DebugDumpOption::PrintPtxasLog)) {
+	printf("%s\n", info_log.data());
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp]: Do we have to use `printf`? I remember from a long time ago that upstream doesn't like it. Not sure if it's still valid:

[https://github.com/pytorch/pytorch/pull/46994](https://github.com/pytorch/pytorch/pull/46994)

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 2]
```diff
@@ -740,30 +780,15 @@ NvrtcFunction nvrtcCompile(
...
-        std::min(effective_max_reg_per_warp / warp_size, max_regs_per_thread);
-
-    max_register_usage += std::to_string(max_register);
-    args.push_back(max_register_usage.c_str());
-  }
-
-  const char* ptxas_opt_level = getenv("PYTORCH_NVFUSER_JIT_OPT_LEVEL");
-  uint32_t jit_opt_level = 0;
+    max_register = static_cast<uint32_t>(
+        std::min(effective_max_reg_per_warp / warp_size, max_regs_per_thread));
```
Comment (shmsong) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp:784]: This isn't part of changes introduced in this PR but the math looks strange to me. Why would we divide `per_warp` by `warp_size`?

**Review by shmsong — APPROVED** [review]
LGTM. Thanks for fixing the CI failure 👍 .

Had a minor question.

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 2 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp:784]: effective_max_reg_per_warp / warp_size --> effective_max_reg_per_thread.

I guess it's the poor naming that makes this confusing.

**Pull request merged**